egs/wsj/s5/steps/nnet3/lstm/train.sh

```diff
@@ -451,6 +453,12 @@ while [ $x -lt $num_iters ]; do
   ilr=$initial_effective_lrate; flr=$final_effective_lrate; np=$num_archives_processed; nt=$num_archives_to_process;
   this_learning_rate=$(perl -e "print (($x + 1 >= $num_iters ? $flr : $ilr*exp($np*log($flr/$ilr)/$nt))*$this_num_jobs);");
 
+  if [ "$adaptive_shrink" == "true" ]; then
+    this_shrink=$(perl -e "print $shrink + ($flr*$this_num_jobs/$this_learning_rate) * (1 - $shrink)")
```

Comment:
Actually, after thinking about this some more, I think it might be better to make it
this_shrink=$(perl -e "print 1.0 - (($this_learning_rate / $ilr)) * (1 - $shrink));")
This makes the shrink setting independent of the initial num-jobs, so that you could change the initial num-jobs without materially affecting the shrinkage rate.